src/main/java/org/plumelib/util/RegexUtil.java

```diff
@@ -332,4 +335,89 @@ private static String regexErrorMessage(String s, int expectedGroups, int actual
   private static int getGroupCount(Pattern p) {
     return p.matcher("").groupCount();
   }
+
+  /**
+   * Return the strings such that any one of the regexes matches it.
+   *
+   * @param strings a collection of strings
+   * @param regexes a collection of regular expressions
+   * @return the strings such that any one of the regexes matches it
+   */
+  public static List<String> anyMatches(
+      Collection<String> strings, Collection<@Regex String> regexes) {
+    List<Pattern> patterns = CollectionsPlume.mapList(Pattern::compile, regexes);
+    List<String> result = new ArrayList<String>(strings.size());
+    for (String s : strings) {
+      for (Pattern p : patterns) {
+        if (p.matcher(s).matches()) {
+          result.add(s);
+          break;
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Return true if every string is matched by at least one regex.
+   *
+   * @param strings a collection of strings
+   * @param regexes a collection of regular expressions
+   * @return true if every string is matched by at least one regex
+   */
+  public static boolean allMatch(Collection<String> strings, Collection<@Regex String> regexes) {
```

Comment:
* `anyMatches`: `any` refers to any regex 
* `noneMatch`: `none` refers to none of the regexes.  
* `allMatch`: `all` refers to all of the **strings**

I find the `all` confusing. It should by `anyMatch`,